app/src/main/java/ch/epfl/skysync/database/tables/FlightTable.kt

```diff
@@ -212,7 +215,38 @@ class FlightTable(db: FirestoreDatabase) :
       }
 
   /**
-   * Add a new flight to the database
+   * Returns the available baskets on a given date and timeslot
+   *
+   * @param localDate The requested day
+   * @param timeslot The requested timeslot
+   */
+  suspend fun getBasketsAvailableOn(localDate: LocalDate, timeslot: TimeSlot): List<Basket> {
+    val dateFilter = Filter.equalTo("date", DateLocalDateConverter.localDateToDate(localDate))
+    val timeslotFilter = Filter.equalTo("timeSlot", timeslot)
+    val flightFilter = Filter.and(dateFilter, timeslotFilter)
+
+    val unavailableBasketsIds: List<String> =
+        query(flightFilter).mapNotNull { flight: Flight -> flight.basket?.id }
+
+    val basketFilter = Filter.notInArray(FieldPath.documentId(), unavailableBasketsIds)
+    return basketTable.query(basketFilter)
+  }
+
+  /**
+   * Returns the all the flights of a given user
+   *
+   * @param id The id of the user
+   */
+  suspend fun getFlightsForUser(id: String): List<Flight> {
+    val flightIds =
+        flightMemberTable.query(Filter.equalTo("userId", id)).map { flightMemberSchema ->
+          flightMemberSchema.flightId
+        }
+    return query(Filter.inArray(FieldPath.documentId(), flightIds))
```

Comment:
`inArray` also has limitations ([doc](https://firebase.google.com/docs/firestore/query-data/queries?_gl=1*7wgdrw*_up*MQ..*_ga*MTQwMzAzMzQyNi4xNzE0Mzg3MzA0*_ga_CW55HF8NVT*MTcxNDM4NzMwMy4xLjAuMTcxNDM4NzMwMy4wLjAuMA..#in_not-in_and_array-contains-any)), here you can also send a query per flight id:
```kotlin
return flightIds.map {id ->
  async { get(id) }
}.awaitAll()
```